core/services/chainlink/application.go

```diff
@@ -297,10 +301,37 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 	promReporter := promreporter.NewPromReporter(db.DB, globalLogger)
 	srvcs = append(srvcs, promReporter)
 
+	// Initialize Local Users ORM and Authentication Provider specified in config
+	// LocalAdminUsersORM is initialized and required regardless of separate Authentication Provider
+	var localAdminUsersORM sessions.LocalAdminUsersORM
+	localAdminUsersORM = localauth.NewORM(db, cfg.WebServer().SessionTimeout().Duration(), globalLogger, cfg.Database(), auditLogger)
+
+	// Initialize Sessions ORM based on environment configured authenticator
+	// localDB auth or remote LDAP auth
+	authMethod := cfg.WebServer().AuthenticationMethod()
+	var authenticationProvider sessions.AuthenticationProvider
+	var sessionReaper utils.SleeperTask
+
+	switch authMethod {
+	case string(sessions.LDAPAuth):
+		var err error
+		authenticationProvider, err = ldapauth.NewLDAPAuthenticator(
+			db, cfg.Database(), cfg.WebServer().LDAP(), cfg.Insecure().DevWebServer(), globalLogger, auditLogger,
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "NewApplication: failed to initialize LDAP Authentication module")
+		}
+		sessionReaper = ldapauth.NewLDAPServerStateSync(db, cfg.Database(), cfg.WebServer().LDAP(), globalLogger)
+	case string(sessions.LocalAuth):
+		authenticationProvider = localauth.NewORM(db, cfg.WebServer().SessionTimeout().Duration(), globalLogger, cfg.Database(), auditLogger)
+		sessionReaper = localauth.NewSessionReaper(db.DB, cfg.WebServer(), globalLogger)
+	default:
+		return nil, errors.Errorf("NewApplication: Unexpected 'AuthenticationMethod': %s supported values: %s, %s", authMethod, string(sessions.LocalAuth), string(sessions.LDAPAuth))
```

Comment:
These shouldn't be required:
```suggestion
		return nil, errors.Errorf("NewApplication: Unexpected 'AuthenticationMethod': %s supported values: %s, %s", authMethod, sessions.LocalAuth, sessions.LDAPAuth)
```

Reply:
Nice, updated